ashley/src/test/kotlin/ktx/ashley/EnginesSpec.kt

```diff
@@ -124,5 +124,16 @@ object EnginesSpec : Spek({
         assertThat(entity.getComponent(Texture::class.java)).isNotNull()
       }
     }
+
+    describe("getSystem function") {
+      it("should add a system and return it") {
+        engine.addSystem(TestSystem())
+        assertThat(engine.getSystem<TestSystem>()).isNotNull()
```

Comment:
Instead of checking if the system is null, you can check if it matches the exact instance of `TestSystem`.

You have only tested the "happy path". You should also add 2 tests for `getSystem` and `get` when the system is absent in the engine.